### PR TITLE
State tracking based on GMCP

### PR DESCRIPTION
**Pull request by RollanzMushing**
Current state of afflictions and defences are now stored in my_affs and my_defs, and updated as relevant GMCP messages are received.

GMCPTrack is a table of functions whose keys correspond to GMCP messages. GMCPTrackProcess(source, message) is the function that actually calls the function if one is available and returns otherwise.

In the future, additional state tracking can be implemented by adding appropriate functions to the GMCPTrack table.

### CODE REVIEW

**Comment by akj** [comment]
These look great, going to review them formally later this weekend and incorporate the gmcp stuff into the oracle namespace.
Also, I should establish standards for this on the wiki but I am personally a fan of camel case for function names. I copied the gmcp code whole cloth so there are _s there. DO you have a reason to use _s in your own functions or is it just for readability?

**Comment by RollanzMushing** [comment]
I use underscores because I typically deal with a lot of abbreviations, which I find hard to parse when mixed with camel case. I have no problem switching to camel case for this project.

**Comment by akj** [comment]
This is outstanding, especially now that gmcp is reporting the stacking affs correctly. Thank you for implementing this @RollanzMushing !
